src/services/bundle.service.js

```diff
@@ -153,24 +157,67 @@ async function uploadTransactionBundle(req, res) {
   return bundle;
 }
 
-async function uploadResourcesFromBundle(entries, headers, baseUrl, baseVersion, protocol, xprovenanceIncluded) {
+async function uploadResourcesFromBundle(entries, headers, xprovenanceIncluded, baseVersion) {
   const scrubbedEntries = replaceReferences(entries);
   // define headers to be included in axios call
   const entryHeaders = { 'Content-Type': 'application/json+fhir' };
   if (xprovenanceIncluded) {
     entryHeaders['X-Provenance'] = headers['x-provenance'];
   }
   const requestsArray = scrubbedEntries.map(async entry => {
-    const { url, method } = entry.request;
-    const destinationUrl = `${protocol}://${path.join(headers.host, baseUrl, baseVersion, url)}`;
-    return axios[method.toLowerCase()](destinationUrl, entry.resource, {
-      headers: entryHeaders
-    }).catch(e => {
-      return e.response;
+    const { method } = entry.request;
+
+    return insertBundleResources(entry, method).catch(e => {
+      const operationOutcome = resolveSchema(baseVersion, 'operationoutcome');
+      const results = new operationOutcome();
+      results.issue = e.issue;
+      results.statusCode = e.statusCode;
+      return {
+        status: e.statusCode,
+        statusCode: e.statusCode,
+        statusText: e.issue[0].code,
+        data: results.toJSON()
+      };
     });
   });
   const requestResults = await Promise.all(requestsArray);
   return requestResults;
 }
 
+async function insertBundleResources(entry, method) {
+  //need to return an array of promises
+  checkSupportedResource(entry.resource.resourceType);
+  if (method === 'POST') {
+    entry.resource.id = uuidv4();
+    const { id } = await createResource(entry.resource, entry.resource.resourceType);
+    if (id != null) {
+      entry.status = 201;
+      entry.statusText = 'Created';
+    }
+  }
+  if (method === 'PUT') {
+    const { id, created } = await updateResource(entry.resource.id, entry.resource, entry.resource.resourceType);
+    if (created === true) {
+      entry.status = 201;
+      entry.statusText = 'Created';
+    } else if (id != null && created === false) {
+      entry.status = 200;
+      entry.statusText = 'OK';
+    }
+  } else {
```

Comment:
I think it's worth adding a unit test with an invalid request type to hit this branch